### PR TITLE
[Merge to Main] Fix/ping 2292 ping 2830

### PR DESCRIPTION
**Pull request by fajarism**
This commits include:
1. Add trigger to refetch data when chat tab is clicked
2. Add some mocks
3. Add test for ChannelListScreenV2

### CODE REVIEW

**Comment by fajarism** [comment]
mas @arilukmanp @agit26milan @iwangsyah  Mbak @shabibsoc  please help on review ya

File: src/hooks/core/chat/useFetchAnonymousChannelHook.ts [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+      console.log('error on getting anonymousChannel');
+      console.log(e);
+    }
+
+    try {
+      await __saveAllAnonymousChannelData(anonymousChannel);
+      refresh('channelList');
+    } catch (e) {
+      console.log('error on saving anonymousChannel');
+      console.log(e);
```
Comment (agit26milan) [inline src/hooks/core/chat/useFetchAnonymousChannelHook.ts:86]: I think this console declare twice

File: src/hooks/core/chat/useFetchAnonymousChannelHook.ts [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+
+  const getAllAnonymousChannels = async () => {
+    if (!localDb) return;
+    let anonymousChannel = [];
+
+    try {
+      anonymousChannel = await AnonymousMessageRepo.getAllAnonymousChannels();
+    } catch (e) {
+      console.log('error on getting anonymousChannel');
+      console.log(e);
```
Comment (agit26milan) [inline src/hooks/core/chat/useFetchAnonymousChannelHook.ts:78]: console nya ada dua

File: src/hooks/core/chat/useFetchAnonymousChannelHook.ts [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+        await userMember.saveOrUpdateIfExists(localDb);
+      });
+
+      channel?.messages?.forEach(async (message) => {
+        const chat = ChatSchema.fromGetAllAnonymousChannelAPI(channel?.id, message);
+        await chat.save(localDb);
+      });
+    } catch (e) {
+      console.log('error on saveAnonymousChannelData');
+      console.log(e);
```
Comment (agit26milan) [inline src/hooks/core/chat/useFetchAnonymousChannelHook.ts:60]: consolenya ada dua

File: src/hooks/core/chat/useFetchAnonymousChannelHook.ts [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+    });
+  };
+
+  const __saveAnonymousChannelData = async (channel) => {
+    if (!channel?.members || channel?.members?.length === 0) return;
+    try {
+      await __helperAnonymousChannelPromiseBuilder(channel);
+    } catch (e) {
+      console.log('error on saveAnonymousChannelData helperAnonymousChannelPromiseBuilder');
+      console.log(e);
```
Comment (agit26milan) [inline src/hooks/core/chat/useFetchAnonymousChannelHook.ts:39]: consolenya dua

**Review by agit26milan — CHANGES_REQUESTED** [review]

File: src/hooks/core/chat/useFetchAnonymousChannelHook.ts (hunk 1 above)
Comment (fajarism) [reply src/hooks/core/chat/useFetchAnonymousChannelHook.ts:86]: These are for clearer message, if we use only `console.log(e)` we don't know which one is logging the error. Unless we use better log framework, we still have to do this.

File: src/hooks/core/chat/useFetchAnonymousChannelHook.ts (hunk 3 above)
Comment (fajarism) [reply src/hooks/core/chat/useFetchAnonymousChannelHook.ts:60]: https://github.com/BetterSocial/mobileapp/pull/1463#discussion_r1354350101

File: src/hooks/core/chat/useFetchAnonymousChannelHook.ts (hunk 4 above)
Comment (fajarism) [reply src/hooks/core/chat/useFetchAnonymousChannelHook.ts:39]: https://github.com/BetterSocial/mobileapp/pull/1463#discussion_r1354350101

File: src/hooks/core/chat/useFetchAnonymousChannelHook.ts (hunk 1 above)
Comment (agit26milan) [reply src/hooks/core/chat/useFetchAnonymousChannelHook.ts:86]: or can use this message mas console.log({error:e}, "error message")

**Review by agit26milan — APPROVED** [review]

**Pull request merged**